source/MRViewer/MRMeshShader.cpp

```diff
@@ -165,7 +166,9 @@ std::string getMeshFragmentShaderColoringBlock()
 
     if ( useTexture && !selected )
     {
-      vec4 textColor = texture(tex, vec3(texcoordi.x, texcoordi.y, 0.0));
+      ivec2 tPFTexSize = textureSize( texturePerFace, 0 );
+      uint textId = texelFetch(texturePerFace, ivec2( primitiveId % uint(tPFTexSize.x), primitiveId / uint(tPFTexSize.x) ), 0 ).r;
```

Comment:
Suppot zero sized texture per face, if there is only one texture, no need to use it